lib/html_parser.dart

```diff
@@ -52,7 +54,9 @@ class HtmlParser extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    dom.Document document = parseHTML(htmlData);
+    dom.Document document;
+    if (userDocument == null) document = parseHTML(htmlData);
```

Comment:
Maybe this is a prime use-case to move the parsing of the HTML to the `Html` widget itself and have this `HtmlParser` widget (despite the name...) just deal with the actual rendering of the DOM? Then this `HtmlParser` widget would only accept an (already parsed) `document`.

Reply:
I agree. This may also help with the bug being discussed about the widget rebuilding a ton when scrolling? Parsing HTML has to be a little expensive especially with the huge document sizes, likely this will help that a bit if just the `HtmlParser` widget is being rebuilt.

Reply:
That'd be fantastic! I wonder still how we can test (ideally even unit test) how often parsing and rendering is done. I'd have to learn more about it to really contribute here.

As for this pr, this is looking good! I'll test run.

Reply:
You could do something pretty simple to debug I think, just put `print("fired")` in the `parseHtm`| function and watch the logs. 

I might do this today just to investigate.